cas-server-core/src/main/java/org/jasig/cas/services/DefaultRegisteredServiceAuthorizationStrategy.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.services;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.jasig.cas.authentication.principal.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.validation.constraints.NotNull;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This is {@link DefaultRegisteredServiceAuthorizationStrategy}
+ * that allows the following rules:
+ *
+ * <ul>
+ *     <li>A service may be disallowed to use CAS for authentication</li>
+ *     <li>A service may be disallowed to take part in CAS single sign-on such that
+ *     presentation of credentials would always be required.</li>
+ *     <li>A service may be prohibited from receiving a service ticket
+ *     if the existing principal attributes don't contain the required attributes
+ *     that otherwise grant access to the service.</li>
+ * </ul>
+ *
+ * @author Misagh Moayyed mmoayyed@unicon.net
+ * @since 4.1
+ */
+public class DefaultRegisteredServiceAuthorizationStrategy implements RegisteredServiceAuthorizationStrategy {
+
+    private static final long serialVersionUID = 1245279151345635245L;
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    /** Is the service allowed at all? **/
+    private boolean enabled = true;
```

Comment:
Except in this case, this has nothing to do with the user.  The user's own information is not the reason they can't log into the service.

A a concrete example, Rutger's online web registration has hours of availability (yes, I know its weird, but mainframes!).  I would find it weird if during the outside of the hours of availability it appeared to the user as if they were the reason they couldn't access it.  I do see what you're attempting to do, and I'm not horribly against it, but I'm also not convinced its right.  It would be useful for others to chime in.